### ISSUE TITLE
[feature] add prefix var to pkg-config alias

### ISSUE THREAD

**Issue by bruchar1** [conan-io/conan#14049]
### What is your suggestion?

When a package contains components, conan pkg-config generator generates one .pc file per component, plus a root .pc file that includes all the components .pc files.

However, the generated root .pc file does not contain the `prefix` variable. When I use the meson build system, I may need to access that variable, since this is the easiest way to get the location of the installed conan package. For instance, I may need this path to find a tool installed along with this pacakge. An other example is the `boost` package. meson expects the `boost.pc` package to have the `prefix` variable, and not having it causes issues (https://github.com/mesonbuild/meson/pull/11487 was an attempt to fix this on the meson side, but the real problem is on the conan generated file, imho).

Another problem with the generated root .pc file is that it doesn't honnor the `pkg_config_custom_content` property. Therefore, it is not possible to inject custom contents in the generated root .pc file.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @bruchar1 

The PR https://github.com/conan-io/conan/pull/17568 has been merged, as a refactor.
This PR removes the ``{prefix}`` exclusively from ``alias`` pc files, but not from the root. We have seen that there are only 3 usages of ``pkg_config_aliases`` in ConanCenter, so it seems this is a very rarely used feature, as today we would have probably not added it, as Conan allows to override ``pkg_config_name`` property on consumers for special cases.

As I understand the above, the main usage for this request and https://github.com/conan-io/conan/pull/14051 was to have it in the "root" pc file. This has not been removed, and it is still there.

Can you please confirm if your use case was about the "root" pc files, or about the aliased (defined with ````pkg_config_aliases``) ones?

**Comment by bruchar1** [comment]
Yes, I think this is correct. The `{prefix}` was mainly required for the integration of the boost package with meson, and it doesn't use `pkg_config_aliases`.

**Comment by memsharded** [comment]
Great, thanks very much for the feedback!
Then we are leaving the refactor https://github.com/conan-io/conan/pull/17568 for next 2.12

**Comment by Talkless** [comment]
Looks like I've hit on similar issue with Gstreamer + OpenCV.

If one wants to build `opencv` plugin for gstreamer, OpenCV's .pc file must have "prefix" to allow Meson to find data files (some .xml configs used for face/hand detection or smth): https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/blob/ca8068c6d793d7aaa6f2e2cc6324fdedfe2f33fa/ext/opencv/meson.build#L87

Currently, `gst-plugins-bad` built gives me:

```
WARNING: pkgconfig variable 'prefix' not defined for dependency opencv.
...
Message: Unable to detect OpenCV data directory
```

`opencv.pc` contents is:

```
Name: opencv
Description: Conan package: opencv
Version: 3.4.20
Requires: opencv-opencv_core opencv-opencv_core_alias opencv-opencv_flann opencv-opencv_flann_alias opencv-opencv_imgproc opencv-opencv_imgproc_alias opencv-opencv_ml opencv-opencv_ml_alias opencv-opencv_photo opencv-opencv_photo_alias opencv-opencv_video opencv-opencv_video_alias opencv-opencv_features2d opencv-opencv_features2d_alias opencv-opencv_imgcodecs opencv-opencv_imgcodecs_alias opencv-opencv_shape opencv-opencv_shape_alias opencv-opencv_videoio opencv-opencv_videoio_alias opencv-opencv_calib3d opencv-opencv_calib3d_alias opencv-opencv_highgui opencv-opencv_highgui_alias opencv-opencv_objdetect opencv-opencv_objdetect_alias opencv-opencv_superres opencv-opencv_superres_alias opencv-opencv_videostab opencv-opencv_videostab_alias opencv-opencv_phase_unwrapping opencv-opencv_phase_unwrapping_alias opencv-opencv_plot opencv-opencv_plot_alias opencv-opencv_reg opencv-opencv_reg_alias opencv-opencv_surface_matching opencv-opencv_surface_matching_alias opencv-opencv_xphoto opencv-opencv_xphoto_alias opencv-opencv_freetype opencv-opencv_freetype_alias opencv-opencv_fuzzy opencv-opencv_fuzzy_alias opencv-opencv_hfs opencv-opencv_hfs_alias opencv-opencv_img_hash opencv-opencv_img_hash_alias opencv-opencv_line_descriptor opencv-opencv_line_descriptor_alias opencv-opencv_saliency opencv-opencv_saliency_alias opencv-opencv_datasets opencv-opencv_datasets_alias opencv-opencv_rgbd opencv-opencv_rgbd_alias opencv-opencv_stereo opencv-opencv_stereo_alias opencv-opencv_structured_light opencv-opencv_structured_light_alias opencv-opencv_tracking opencv-opencv_tracking_alias opencv-opencv_xfeatures2d opencv-opencv_xfeatures2d_alias opencv-opencv_ximgproc opencv-opencv_ximgproc_alias opencv-opencv_xobjdetect opencv-opencv_xobjdetect_alias opencv-opencv_aruco opencv-opencv_aruco_alias opencv-opencv_bgsegm opencv-opencv_bgsegm_alias opencv-opencv_bioinspired opencv-opencv_bioinspired_alias opencv-opencv_ccalib opencv-opencv_ccalib_alias opencv-opencv_dpm opencv-opencv_dpm_alias opencv-opencv_face opencv-opencv_face_alias opencv-opencv_optflow opencv-opencv_optflow_alias opencv-numeric opencv-opencv_stitching opencv-opencv_stitching_alias opencv-multiview opencv-correspondence opencv-opencv_sfm opencv-opencv_sfm_alias
```
Other `opencv-opencv_*.pc` does have `prefix` in them..

@memsharded Is there any workarounds to achieve this (add `prefix` in `opencv.pc`) in deprecated 1.66..? (haven't migrated to Conan 2 yet...)
